formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Decoder.kt

```diff
@@ -0,0 +1,636 @@
+        val size = readNumber().toInt()
+        readByte()
+        return size
+    }
+
+    fun isEnd() = curByte == BREAK
+
+    fun end() = skipByte(BREAK)
+
+    fun nextByteString(tags: ULongArray? = null): ByteArray {
+        processTags(tags)
+        if ((curByte and 0b111_00000) != HEADER_BYTE_STRING)
+            throw CborDecodingException("start of byte string", curByte)
+        val arr = readBytes()
+        readByte()
+        return arr
+    }
+
+    fun nextString(tag: ULong) = nextString(ulongArrayOf(tag))
+    fun nextString(tags: ULongArray? = null) = nextTaggedString(tags).first
+
+    //used for reading the tag names and names of tagged keys (of maps, and serialized classes)
+    private fun nextTaggedString(tags: ULongArray?): Pair<String, ULongArray?> {
+        val collectedTags = processTags(tags)
+        if ((curByte and 0b111_00000) != HEADER_STRING)
+            throw CborDecodingException("start of string", curByte)
+        val arr = readBytes()
+        val ans = arr.decodeToString()
+        readByte()
+        return ans to collectedTags
+    }
+
+    private fun readBytes(): ByteArray =
+        if (curByte and 0b000_11111 == ADDITIONAL_INFORMATION_INDEFINITE_LENGTH) {
+            readByte()
+            readIndefiniteLengthBytes()
+        } else {
+            val strLen = readNumber().toInt()
+            input.readExactNBytes(strLen)
+        }
+
+    @SuppressAnimalSniffer
+    private fun processTags(tags: ULongArray?): ULongArray? {
+        var index = 0
+        val collectedTags = mutableListOf<ULong>()
+        while ((curByte and 0b111_00000) == HEADER_TAG) {
+            val readTag = readNumber().toULong() // This is the tag number
+            collectedTags += readTag
+            // value tags and object tags are intermingled (keyTags are always separate)
+            // so this check only holds if we verify both
+            if (verifyObjectTags) {
+                tags?.let {
+                    if (index++ > it.size) throw CborDecodingException("More tags found than the ${it.size} tags specified")
+                    if (readTag != it[index - 1]) throw CborDecodingException("CBOR tag $readTag does not match expected tag $it")
+                }
+            }
+            readByte()
+        }
+        return (if (collectedTags.isEmpty()) null else collectedTags.toULongArray()).also { collected ->
+            //We only want to compare if tags are actually set, otherwise, we don't care
+            tags?.let {
+                if (verifyObjectTags) { //again, this check only works if we verify value tags and object tags
+                    if (!it.contentEquals(collected))
+                        throw CborDecodingException(
+                            "CBOR tags ${
+                                collected?.joinToString(
+                                    prefix = "[",
+                                    postfix = "]"
+                                ) { it.toString() }
```

Comment:
`it.toString()` is marked by animal sniffer, apply `@SuppressAnimalSniffer`

Reply:
Ah, I see it is on the function already. Then it is misconfigured. I guess this place: https://github.com/Kotlin/kotlinx.serialization/blob/5aa10324cc652a1f3f9621ec3e063046a5d356be/buildSrc/src/main/kotlin/animalsniffer-conventions.gradle.kts#L55 was lost during rebase on KTS migration

Reply:
This annotation is not applicable to expressions and the method already has the annotation. what should I do?

Reply:
tried a workaround in ac17821b77e25dc2e5316bb07ba156f545a8fd0f

Reply:
Check my comment above — add annotation to `[animalsniffer-conventions.gradle.kts](https://github.com/Kotlin/kotlinx.serialization/blob/5aa10324cc652a1f3f9621ec3e063046a5d356be/buildSrc/src/main/kotlin/animalsniffer-conventions.gradle.kts#L55)`

Reply:
Sorry, GitHub didn't refresh, apparently. Added the line in 3724fea9e39ef3f44cf2a160f3e602cec41459ac and successfully published to a local repo, for what it's worth.

Reply:
It seems that the build is still failing. I've poked around a little bit, and it seems that annotation does not affect the `processTags` function for some reason (probably because the problematic method is inside a lambda). So you have to apply `@SuppressAnimalSniffer` to the whole `class CborParser`

Reply:
done in eb6ac40d6f429144b030d66efb8718286b1c4b3c